src/kani-verifier/src/lib.rs

```diff
@@ -70,6 +71,24 @@ fn appears_setup() -> bool {
     kani_dir().exists()
 }
 
+/// In dev environments, this proxy shouldn't be used.
+/// But accidentally using it (with the test suite) can fire off
+/// hundreds of HTTP requests trying to download a non-existent release bundle.
+/// So if we positively detect a dev environment, raise an error early.
+fn fail_if_in_dev_environment() -> Result<()> {
+    if let Ok(exe) = std::env::current_exe() {
+        if let Some(path) = exe.parent() {
+            if path.ends_with("target/debug") || path.ends_with("target/release") {
+                bail!(
+                    "Running the proxies from a development environment? These are just meant for releases."
```

Comment:
Can we clarify the error message a bit? Example: "Running a release-only executable, \<executable-name\>, from a development environment. Please remove executable from path and rerun."